packages/components/pages/f-mfa/src/services/_tests/AccountWeb.api.test.js

```diff
@@ -0,0 +1,82 @@
+import AccountWebApi from '../AccountWeb.api';
+
+import {
+    SUBMIT_LOGIN_CHALLENGE_URL,
+    CONVERSATION_ID_HEADER_NAME,
+    CONVERSATION_ID_COOKIE_NAME
+} from '../../constants';
+import {
+    baseUrl,
+    conversationId,
+    postBody
+} from '../../../test-utils/setup';
+
+let apiProvider;
+let httpClientMock;
+let cookiesMock;
+const cookiesSetSpy = jest.fn();
+const httpPostSpy = jest.fn();
+
+describe('AccountWebApi Provider', () => {
+    beforeEach(() => {
+        httpClientMock = {
+            post: httpPostSpy
+        };
+        cookiesMock = {
+            set: cookiesSetSpy
+        };
+
+        apiProvider = new AccountWebApi({ httpClient: httpClientMock, cookies: cookiesMock, baseUrl });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('When creating a new instance', () => {
+        it('should not throw error when instance is created with valid parameters', () => {
+            // Act
+            const createInstance = () => new AccountWebApi({
+                httpClient: {}
+            });
+
+            // Assert
+            let instance;
+            expect(() => { instance = createInstance(); }).not.toThrowError();
+            expect(instance).toBeDefined();
+        });
+    });
+
+    describe('When calling `postValidateMfaToken`', () => {
+        it('should send the correct parameters to the api', async () => {
+            // Arrange
+            const expectedUri = `${baseUrl}/${SUBMIT_LOGIN_CHALLENGE_URL}`;
+            const expectedBody = postBody;
+            const expectedHeaders = {
+                [CONVERSATION_ID_HEADER_NAME]: conversationId
+            };
+
+            // Act
+            await apiProvider.postValidateMfaToken(postBody, conversationId);
+
+            // Assert
+            expect(httpPostSpy).toHaveBeenCalledWith(expectedUri, expectedBody, expectedHeaders);
+        });
+
+        it('should set the cookie with a new conversation id if not provided with one', async () => {
+            // Act
+            await apiProvider.postValidateMfaToken(postBody);
+
+            // Assert
+            expect(cookiesSetSpy).toHaveBeenCalledWith(CONVERSATION_ID_COOKIE_NAME, expect.anything());
```

Comment:
How comes this is `expect.anything()`? Shouldn't it be at least expecting a string or some sort of formatted expectation?

Reply:
@kevinrodrigues  Because this test is to test the conversation id is being created (i.e. calling the cookie.set()) and nothing else plus it is very difficult to pre-determine the random guid that will be created for the conversation id ... plus not that revealing even if we could.

Reply:
To me (and could just be me) looks like the test is checking to make sure cookieSetSpy is:
1. Being called correctly.
2. Is being called with the right inputs.

The correct inputs was my worry as it should reflect the expected type. The coversation ID i suspect would be a unique id generated via a string so this test would only need to change to expect.any(String)`. Happy to leave as is tho.

Reply:
Sure, will change to `expect.any(String)` on next pr 👍